library/libobject.ml

```diff
@@ -16,11 +16,20 @@ module Dyn = Dyn.Make ()
 type 'a substitutivity =
     Dispose | Substitute of 'a | Keep of 'a | Anticipate of 'a
 
+type categories = Names.Id.t list
+
+let filter_cat ocat f ~cat i o =
```

Comment:
It is a silly detail given the size of this PR, but `filter_cat` is a bad name since `cat` is another name for `append` in some functional languages, so I'm suggesting `filter_by_category`